### ISSUE TITLE
Update EAP addon to have correct EAP templates

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#653]
Hi,

I am just wondering why EAP templates doesn't have templates with no database for general templates. 
Are we are only providing persistent related templates?

When I do `oc get templates -n openshift`, I am getting following: 
```
oc get templates -n openshift
NAME                                DESCRIPTION                                                                        PARAMETERS        OBJECTS
cakephp-mysql-persistent            An example CakePHP application with a MySQL database. For more information ab...   20 (4 blank)      9
dancer-mysql-persistent             An example Dancer application with a MySQL database. For more information abo...   17 (5 blank)      9
django-psql-persistent              An example Django application with a PostgreSQL database. For more informatio...   18 (5 blank)      9
jenkins-ephemeral                   Jenkins service, without persistent storage....                                    6 (all set)       6
jenkins-persistent                  Jenkins service, with persistent storage....                                       7 (all set)       7
jenkins-pipeline-example            This example showcases the new Jenkins Pipeline integration in OpenShift,...       16 (4 blank)      8
logging-deployer-account-template   Template for creating the deployer account and roles needed for the aggregate...   0 (all set)       9
logging-deployer-template           Template for running the aggregated logging deployer in a pod. Requires empow...   35 (18 blank)     1
mariadb-persistent                  MariaDB database service, with persistent storage. For more information about...   8 (3 generated)   4
mongodb-persistent                  MongoDB database service, with persistent storage. For more information about...   9 (3 generated)   4
mysql-persistent                    MySQL database service, with persistent storage. For more information about u...   9 (3 generated)   4
nodejs-mongo-persistent             An example Node.js application with a MongoDB database. For more information...    17 (4 blank)      9
postgresql-persistent               PostgreSQL database service, with persistent storage. For more information ab...   8 (2 generated)   4
rails-pgsql-persistent              An example Rails application with a PostgreSQL database. For more information...   21 (4 blank)      9
```

It seems that CDK 2* had templates with no persistent, : 
```
### CDK 2.4
oc get templates -n openshift
NAME                                 DESCRIPTION                                                                        PARAMETERS      OBJECTS
cakephp-example                      An example CakePHP application with no database. For more information about u...   13 (4 blank)    6
cakephp-mysql-example                An example CakePHP application with a MySQL database. For more information ab...   19 (4 blank)    8
eap64-basic-s2i                      Application template for EAP 6 applications built using S2I.                       13 (3 blank)    5
eap64-mysql-persistent-s2i           Application template for EAP 6 MySQL applications with persistent storage bui...   37 (17 blank)   10
jenkins-ephemeral                    Jenkins service, without persistent storage....                                    6 (all set)     6
jenkins-persistent                   Jenkins service, with persistent storage....                                       7 (all set)     7
jws30-tomcat7-mysql-persistent-s2i   Application template for JWS MySQL applications with persistent storage built...   28 (11 blank)   10
nodejs-example                       An example Node.js application with no database. For more information about u...   10 (4 blank)    5
nodejs-mongodb-example               An example Node.js application with a MongoDB database. For more information...    16 (4 blank)    8
```

**Comment by hferentschik** [comment]
@budhrg I am not sure what you mean. an we just close this issue. EAP templates are not ready to use yet and there is an open issue for that.

**Comment by hferentschik** [comment]
Upstream tasks to track https://issues.jboss.org/browse/CDK-92. Whether we will add the eap addon just for CDK or whether we keep it in Minishift needs to be discussed.

**Comment by hferentschik** [comment]
This is related as well https://issues.jboss.org/browse/CDK-109

**Comment by budhrg** [comment]
We seems to have templates for ADB here https://github.com/projectatomic/adb-utils/tree/master/services/openshift/templates/adb

**Comment by jorgemoralespou** [comment]
@budhrg yes, by default oc installs the templates for persistent use cases, although I've raised that already.

@hferentschik templates are installed by default when you use the oc cluster. If you use the origin will install origin based templates, if you use the ocp client will install enterprise based templates.

cc/ @csrwng

**Comment by hferentschik** [comment]
Right, we need to double check what gets installed and sync what's needed for the CDK use case. AFAIU OCP cluster up does not include the EAP templates. At least that's what got reported via CDK-109.

**Comment by praveenkumar** [comment]
> AFAIU OCP cluster up does not include the EAP templates. At least that's what got reported via CDK-109.

No, It doesn't. OCP and oc have same template but only imagestream change from centos to rhel AFAIK.

**Comment by jorgemoralespou** [comment]
@praveenkumar seem correct (https://github.com/openshift/origin/blob/master/pkg/bootstrap/docker/up.go#L95-L118)

@csrwng Shouldn't the templates for ocp need to be the ocp ones and not just the imagestreams in oc cluster?

**Comment by csrwng** [comment]
@jorgemoralespou shouldn't the templates be the same and just the imagestreams point to different images?

**Comment by jorgemoralespou** [comment]
@csrwng It's missing all the product templates for ocp, that is xPaaS
(eap,...)

**Comment by csrwng** [comment]
@jorgemoralespou I see what you mean. Yes we're not including those currently.

**Comment by jorgemoralespou** [comment]
Should we? I'm not in favor and since most enterprise users are using
oc-cluster to add then via a plugin I'm fine, but I think this would need
to be discussed with Ben and Clayton.

**Comment by praveenkumar** [comment]
@jorgemoralespou Doesn't that addon feature of minishift going to address it or we need it to be part of default oc cluster up experience?

**Comment by hferentschik** [comment]
>  Doesn't that addon feature of minishift going to address it 

Sure, that is our solution for now, especially for the CDK

> or we need it to be part of default oc cluster up experience?

note really, but it would for sure be nice

**Comment by LalatenduMohanty** [comment]
> Doesn't that addon feature of minishift going to address it or we need it to be part of default oc cluster up experience?

One thing to keep in mind that, there will be users of `oc cluster up` too. So if something adds vale to `oc cluster up` as well as Minishift we should add it to `oc cluster up` and then Minishift take advantage of it. This way everyone wins.

**Comment by jorgemoralespou** [comment]
@praveenkumar as @LalatenduMohanty mentions, there are some features that will belong to 'oc cluster' and will be implemented there. It's just a matter of aligning both efforts. :-D

@csrwng whenever you can, please give us an answer if this will ever be (or soonish) on "oc cluster". For the time being, as in 1.5.0 will not exist, the addon is the way for now.

**Comment by hferentschik** [comment]
Merged via pull request #772
